test/acceptance/tests/hcp/terraform/ns/main.tf

```diff
@@ -0,0 +1,264 @@
+}
+
+resource "consul_namespace" "ns2" {
+  name        = "ns2"
+  description = "Namespace 2"
+}
+
+resource "aws_secretsmanager_secret" "bootstrap_token" {
+  name = "${var.suffix}-bootstrap-token"
+}
+
+resource "aws_secretsmanager_secret_version" "bootstrap_token" {
+  secret_id     = aws_secretsmanager_secret.bootstrap_token.id
+  secret_string = hcp_consul_cluster.this.consul_root_token_secret_id
+}
+
+resource "aws_secretsmanager_secret" "gossip_key" {
+  name = "${var.suffix}-gossip-key"
+}
+
+resource "aws_secretsmanager_secret_version" "gossip_key" {
+  secret_id     = aws_secretsmanager_secret.gossip_key.id
+  secret_string = jsondecode(base64decode(hcp_consul_cluster.this.consul_config_file))["encrypt"]
+}
+
+resource "aws_secretsmanager_secret" "consul_ca_cert" {
+  name = "${var.suffix}-consul-ca-cert"
+}
+
+resource "aws_secretsmanager_secret_version" "consul_ca_cert" {
+  secret_id     = aws_secretsmanager_secret.consul_ca_cert.id
+  secret_string = base64decode(hcp_consul_cluster.this.consul_ca_file)
+}
+
+// Create ACL controller.
+module "acl_controller" {
+  source = "../../../../../../modules/acl-controller"
+  log_configuration = {
+    logDriver = "awslogs"
+    options = {
+      awslogs-group         = var.log_group_name
+      awslogs-region        = var.region
+      awslogs-stream-prefix = "consul-acl-controller"
+    }
+  }
+  launch_type                       = var.launch_type
+  consul_bootstrap_token_secret_arn = aws_secretsmanager_secret.bootstrap_token.arn
+  consul_server_http_addr           = hcp_consul_cluster.this.consul_private_endpoint_url
+  ecs_cluster_arn                   = var.ecs_cluster_arn
+  region                            = var.region
+  subnets                           = var.subnets
+  name_prefix                       = var.suffix
+  consul_ecs_image                  = var.consul_ecs_image
+}
+
+// Create services.
+resource "aws_ecs_service" "test_client" {
+  name            = "test_client_${var.suffix}"
+  cluster         = var.ecs_cluster_arn
+  task_definition = module.test_client.task_definition_arn
+  desired_count   = 1
+  network_configuration {
+    subnets = var.subnets
+  }
+  launch_type            = var.launch_type
+  propagate_tags         = "TASK_DEFINITION"
+  enable_execute_command = true
+
+  tags = var.tags
+}
+
+module "test_client" {
+  source = "../../../../../../modules/mesh-task"
+  family = "test_client_${var.suffix}"
+  container_definitions = [{
+    name      = "basic"
+    image     = "docker.mirror.hashicorp.services/nicholasjackson/fake-service:v0.21.0"
+    essential = true
+    environment = [
+      {
+        name  = "UPSTREAM_URIS"
+        value = "http://localhost:1234"
+      }
+    ]
+    linuxParameters = {
+      initProcessEnabled = true
+    }
+  }]
+  retry_join = jsondecode(base64decode(hcp_consul_cluster.this.consul_config_file))["retry_join"]
+  upstreams = [
+    {
+      destinationName = "test_server_${var.suffix}"
```

Comment:
In this upstream config for the client, you should need `destinationNamespace = local.test_server_ns` here. This routes the requests from the client service to the test server in that separate namespace.

Reply:
Good catch, thanks! I've updated per the comment.